carla/evaluation/benchmark.py

```diff
@@ -0,0 +1,114 @@
+import csv
+from typing import Dict
+
+import pandas as pd
+
+from carla.evaluation.distances import get_distances
+from carla.models.api import MLModel
+from carla.models.pipelining import encode, scale
+from carla.recourse_methods.api import RecourseMethod
+
+
+class Benchmark:
+    def __init__(
+        self, mlmodel: MLModel, recmodel: RecourseMethod, factuals: pd.DataFrame
+    ) -> None:
+        """
+        Constructor for benchmarking class
+
+        Parameters
+        ----------
+        mlmodel: MLModel
+            Black Box model we want to explain
+        recmodel: RecourseMethod
+            Recourse method we want to benchmark
+        factuals: pd.DataFrame
+            Instances we want to find counterfactuals
+        """
+        self._recmodel = recmodel
+        self._counterfactuals = self._recmodel.get_counterfactuals(factuals)
+
+        # Normalizing and encoding factual for later use
+        self._factuals = scale(mlmodel.scaler, mlmodel.data.continous, factuals)
+        self._factuals = encode(
+            mlmodel.encoder, mlmodel.data.categoricals, self._factuals
+        )
+        self._factuals = self._factuals[
+            mlmodel.feature_input_order + [mlmodel.data.target]
+        ]
+
+    def compute_distances(self) -> Dict:
+        """
+        Calculates the distance measure and returns it as dict
+
+        Returns
+        -------
+        Dict
+        """
+        key = "Distances"
+        output: Dict = {key: []}
+        arr_f = self._factuals.to_numpy()
+        arr_cf = self._counterfactuals.to_numpy()
+
+        distances = get_distances(arr_f, arr_cf)
+        sub_keys = ["d1", "d2", "d3", "d4"]
+        for i in range(len(distances)):
+            output[key].append(dict(zip(sub_keys, distances[i])))
```

Comment:
Why did you choose to (somewhat complicatly) create a dict first and later convert it to csv and not generate a dataframe in the first place

Reply:
I wanted to have some readable output, but haven't thought of DataFrames.
This would make much more sense and would be helpful for later visualization tasks. Will rewrite this in the next commits!